acra-censor/common/common.go

```diff
@@ -150,3 +176,142 @@ func ParsePatterns(rawPatterns []string) ([]sqlparser.Statement, error) {
 	}
 	return outputPatterns, nil
 }
+
+// TrimStringToN trims query to N chars.
+func TrimStringToN(query string, n int) string {
+	if len(query) <= n {
+		return query
+	}
+	return query[:n]
+}
+
+// HandleRawSQLQuery returns a normalized (lowercases SQL commands) SQL string,
+// and redacted SQL string with the params stripped out for display.
+// Taken from sqlparser package
+func HandleRawSQLQuery(sql string) (normalizedQuery, redactedQuery string, parsedQuery sqlparser.Statement, err error) {
+	bv := map[string]*querypb.BindVariable{}
+	sqlStripped, _ := sqlparser.SplitMarginComments(sql)
+
+	// sometimes queries might have ; at the end, that should be stripped
+	sqlStripped = strings.TrimSuffix(sqlStripped, ";")
+
+	stmt, err := sqlparser.Parse(sqlStripped)
+	outputStmt, _ := sqlparser.Parse(sqlStripped)
+	if err != nil {
+		return "", "", nil, ErrQuerySyntaxError
+	}
+
+	normalizedQ := sqlparser.String(stmt)
+
+	// redact and mask VALUES
+	sqlparser.Normalize(stmt, bv, ValueMask)
+	redactedQ := sqlparser.String(stmt)
+
+	return normalizedQ, redactedQ, outputStmt, nil
+}
+
+func CheckPatternsMatching(patterns []sqlparser.Statement, parsedQuery sqlparser.Statement) (bool, error) {
+	for _, pattern := range patterns {
+		if checkSinglePatternMatch(parsedQuery, pattern) {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func CheckExactQueriesMatch(normalizedQuery string, setOfQueries map[string]bool) bool {
+	if !setOfQueries[normalizedQuery] {
+		return false
+	}
+	return true
+}
+
+func CheckTableNamesMatch(parsedQuery sqlparser.Statement, setOfTables map[string]bool) (bool, bool) {
+	atLeastOneTableNameMatch := false
+	allTableNamesMatch := false
+
+	switch parsedQuery.(type) {
```

Comment:
To avoid casts inside `case` statements we can cast once using switch statement and reduce lines of code below
```
switch query := parsedQuery.(type) {
  case *sqlparser.Select:
      // query here casted to *sqlparser.Select
      //query.From
}
```

Reply:
Agree